README.md

```diff
@@ -0,0 +1,65 @@
+# kubernetes-nmstate
+
+Node-networking configuration driven by Kubernetes and executed by
+[nmstate](https://nmstate.github.io/).
+
+## The "Why"
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
+The [Container Network Interface](https://github.com/containernetworking/cni)
+(CNI) standard enables different
+solutions for connecting networks on the node with pods. Some of them are
+[part of the standard](https://github.com/containernetworking/plugins), and there are
+others that extend support for [Open vSwitch bridges](https://github.com/kubevirt/ovs-cni),
+[SR-IOV](https://github.com/hustcat/sriov-cni), and more...
+
+However, in all of these cases, the node must have the networks setup before the
+pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster,
+with dynamic networking requirements, is a challenge by itself - and this is
+what this project is addressing.
+
+## The "How"
+
+We use [nmstate](https://nmstate.github.io/) to perform state driven network
+configuration on each node, as well as to return back its current state.
+The system defines 1 objects, `NodeNetworkState`  that
+is used to report and configure cluster node networks.
+
+For more information, please check
+[kubernetes-nmstate design document](https://docs.google.com/document/d/1282BcYjYGIIxQKgMYi3nQodB4ML_gw9BSs5AXB7QUtg/),
+[developer guide](docs/developer-guide.md) and
+[deployment and usage section](#deployment-and-usage).
+
+### Node Network State
+
+Node Network State objects are created per each node in the cluster and can be
+used to configure and report available interfaces and network configuration.
+
+#### Examples
+
+Read `NodeNetworkState` of node `node01` to list available interfaces and their
+configuration:
+
+```bash
+curl https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/docs/demos/state.yaml
+```
+
```

Comment:
Another example of `NodeNetworkState`, this time specifying desired configuration on the node: ...

Reply:
Well for that we already have the user guide.